packages/remix/src/client/uiComponents.tsx

```diff
@@ -38,17 +39,18 @@ export const CreateOrganization = (props: CreateOrganizationProps) => {
 // "The inferred type of 'OrganizationProfile' cannot be named without a reference to ..."
 export const OrganizationProfile: typeof BaseOrganizationProfile = Object.assign(
   (props: OrganizationProfileProps) => {
-    return <BaseOrganizationProfile {...useRoutingProps('OrganizationProfile', props)} />;
+    const { pathname: path } = useLocation();
+    return <BaseOrganizationProfile {...useRoutingProps('OrganizationProfile', props, { path })} />;
   },
   { ...BaseOrganizationProfile },
 );
 
 export const SignIn = (props: SignInProps) => {
-  const { signInUrl } = useClerkRemixOptions();
-  return <BaseSignIn {...useRoutingProps('SignIn', props, { path: signInUrl })} />;
+  const { pathname: path } = useLocation();
+  return <BaseSignIn {...useRoutingProps('SignIn', props, { path })} />;
 };
 
 export const SignUp = (props: SignUpProps) => {
-  const { signUpUrl } = useClerkRemixOptions();
-  return <BaseSignUp {...useRoutingProps('SignUp', props, { path: signUpUrl })} />;
+  const { pathname: path } = useLocation();
+  return <BaseSignUp {...useRoutingProps('SignUp', props, { path })} />;
```

Comment:
This looks like it overrides the env vars passed in the project. Are we ok with this?

Reply:
Yeah as @nikosdouvlis said before [here](https://github.com/clerk/javascript/pull/2634#discussion_r1458153742)
>I don't think we need to respect this - the env variable is supposed to be used so the SignIn component (and the middleware) knows where SignUp is mounted and vice versa.